### DISCUSSION TITLE
Hidden field for first item in a Repeater

### DISCUSSION THREAD

**Discussion by GregoireGaonach** [filamentphp/filament#2594] [Help] [answered]
Good evening,

I am building a form to do an advanced search with a Repeater.

I have two repeated fields:
- A select `operator` with options `OR` and `AND`.
- A select `filter` with filtered values.

I would like to hide the field `operator` if it is in the **first position** in the Repeater (to have something like `key:filter AND key:filter` rather than `AND key:filter AND key:filter`).

I know there is a `->hidden` option that would hide the field but I'm having trouble finding the right condition.

I think I should compare the UID of the first field in the Repeater with the UID of the current field. I am struggling getting these values.

Do you have any suggestions?

**Comment by danharrin** [comment, marked as answer]
Hmm, how about something like...

```php
->hidden(function (Select $component) {
    $firstItemKey = array_key_first($component->getContainer()->getParentComponent()->getState());

    return str_contains($component->getStatePath(), $firstItemKey);
})
```

**Reply by GregoireGaonach** [reply]
Incredible @danharrin, this is exactly what I was looking for.

Is there a documentation with all these methods (`getContainer()`, `getParentComponent()`, etc.)? I didn't find anything before asking this question.

**Reply by danharrin** [reply]
Generally speaking, this technique is called "closure customisation" - https://filamentphp.com/docs/2.x/forms/advanced#using-closure-customisation

But some of the specific methods I used are not documented, they're internal and very specific, hardly anyone uses them.

The goal is to return a bool from the Closure, true or false, based on if the field should be hidden.

`$component` is the Select component. You can get the Repeater that the Select belongs to, using `$component->getContainer()->getParentComponent()`.

Using the Repeater component, you can get the state (array of repeater data) using `->getState()`.

From the repeater data, you can get the key of the first repeater item using `array_key_first()`, we use this later.

`$component->getStatePath()` is the unique ID of the Select in the form, and it contains the key of the repeater item it is in. Looks something like `data.repeater.1234.select_name`

So then we just use `str_contains()` to check if the Select is in the first repeater item or not :)
